### PR TITLE
docs: Initial HCP Rebrand 

### PR DESCRIPTION
**Pull request by boruszak**
### Description

This PR updates references to Terraform Cloud in the Consul docs.

Do not merge before 9 AM PST on April 22, 2024.

### PR Checklist

* [ ] updated test coverage
* [X] external facing docs updated
* [X] appropriate backport labels added
* [X] not a security concern

### CODE REVIEW

File: website/content/docs/nia/enterprise/index.mdx [hunk 1]
```diff
@@ -9,22 +9,22 @@ description: >-
...
+| Automation Driver | Terraform Community Edition | Terraform Community Edition, HCP Terraform, or Terraform Enterprise |
+| Terraform Workspaces | Local | Local workspaces with the Terraform driver or [remote workspaces](/terraform/cloud-docs/workspaces) with the HCP Terraform driver |
+| Terraform Backend Options | [azurerm](/terraform/language/settings/backends/azurerm), [consul](/terraform/language/settings/backends/consul), [cos](/terraform/language/settings/backends/cos), [gcs](/terraform/language/settings/backends/gcs), [kubernetes](/terraform/language/settings/backends/kubernetes), [local](/terraform/language/settings/backends/local), [manta](/terraform/language/v1.2.x/settings/backends/manta), [pg](/terraform/language/settings/backends/pg), and [s3](/terraform/language/settings/backends/s3) with the Terraform driver | The supported backends for CTS with the Terraform driver or HCP Terraform with the HCP Terraform driver |
+| Terraform Version | One Terraform version for all tasks | Optional Terraform version per task when using the HCP Terraform driver |
+| Terraform Run Output | CTS logs | CTS logs or Terraform output organized by THCP Terraform remote workspaces |
```
Comment (rkoron007) [inline website/content/docs/nia/enterprise/index.mdx:21]:
```suggestion
| Terraform Run Output | CTS logs | CTS logs or Terraform output organized by HCP Terraform remote workspaces |
```

File: website/content/docs/nia/network-drivers/index.mdx [hunk 2]
```diff
@@ -11,18 +11,18 @@ Consul-Terraform-Sync (CTS) uses network drivers to execute and update network i
...
+| [HCP Terraform driver](/consul/docs/nia/network-drivers/terraform-cloud) | CTS Enterprise automates remote workspaces on [HCP Terraform](/terraform/cloud-docs) | - [Remote Terraform execution](/terraform/cloud-docs/run/remote-operations) <br/> - Concurrent runs <br/> - [Secured variables](/terraform/cloud-docs/workspaces/variables) <br/> - [State versions](/terraform/cloud-docs/workspaces/state) <br/> - [Sentinel](/terraform/cloud-docs/policy-enforcement) to enforce governance policies as code <br/> - Audit [logs](/terraform/enterprise/admin/infrastructure/logging) and [trails](/terraform/cloud-docs/api-docs/audit-trails) <br/> - Run [history](/terraform/cloud-docs/run/manage), [triggers](/terraform/cloud-docs/workspaces/settings/run-triggers), and [notifications](/terraform/cloud-docs/workspaces/settings/notifications) <br/> - [HCP Terraform Agents](/terraform/cloud-docs/agents) |
```
Comment (rkoron007) [inline website/content/docs/nia/network-drivers/index.mdx:19]: So unfortunately, Terraform Cloud agents aren't rebranding to HCP Terraform Agents until 5/1. Would you mind removing and doing as a fast follow?
```suggestion
| [HCP Terraform driver](/consul/docs/nia/network-drivers/terraform-cloud) | CTS Enterprise automates remote workspaces on [HCP Terraform](/terraform/cloud-docs) | - [Remote Terraform execution](/terraform/cloud-docs/run/remote-operations) <br/> - Concurrent runs <br/> - [Secured variables](/terraform/cloud-docs/workspaces/variables) <br/> - [State versions](/terraform/cloud-docs/workspaces/state) <br/> - [Sentinel](/terraform/cloud-docs/policy-enforcement) to enforce governance policies as code <br/> - Audit [logs](/terraform/enterprise/admin/infrastructure/logging) and [trails](/terraform/cloud-docs/api-docs/audit-trails) <br/> - Run [history](/terraform/cloud-docs/run/manage), [triggers](/terraform/cloud-docs/workspaces/settings/run-triggers), and [notifications](/terraform/cloud-docs/workspaces/settings/notifications) <br/> - [Terraform Cloud Agents](/terraform/cloud-docs/agents) |
```

File: website/content/docs/nia/network-drivers/terraform-cloud.mdx [hunk 3]
```diff
@@ -103,36 +103,36 @@ You can view the latest service information in the Terraform UI by navigating to
...
 
-1. Hostname of your Terraform Cloud, self-hosted distribution
+1. Hostname of your HCP Terraform, self-hosted distribution
 1. Name of your organization
-1. [Team API token](/terraform/cloud-docs/users-teams-organizations/api-tokens) used for authentication with Terraform Cloud
+1. [Team API token](/terraform/cloud-docs/users-teams-organizations/api-tokens) used for authentication with HCP Terraform
 
-Prior to running CTS with a Terraform Cloud driver, you will need an account and organization set up, as well as a dedicated token. We recommend using a team token that is restricted to [Manage Workspaces](/terraform/cloud-docs/users-teams-organizations/teams#managing-workspace-access)-level permissions. Below are the steps for the recommended setup.
+Prior to running CTS with a HCP Terraform driver, you will need an account and organization set up, as well as a dedicated token. We recommend using a team token that is restricted to [Manage Workspaces](/terraform/cloud-docs/users-teams-organizations/teams#managing-workspace-access)-level permissions. Below are the steps for the recommended setup.
```
Comment (rkoron007) [inline website/content/docs/nia/network-drivers/terraform-cloud.mdx:120]:
```suggestion
Prior to running CTS with an HCP Terraform driver, you will need an account and organization set up, as well as a dedicated token. We recommend using a team token that is restricted to [Manage Workspaces](/terraform/cloud-docs/users-teams-organizations/teams#managing-workspace-access)-level permissions. Below are the steps for the recommended setup.
```

File: website/data/docs-nav-data.json [hunk 4]
```diff
@@ -2056,7 +2056,7 @@
             "path": "nia/enterprise/license"
           },
           {
-            "title": "Terraform Cloud Driver",
+            "title": "HCP Terraform Driver",
             "href": "/docs/nia/network-drivers/terraform-cloud"
```
Comment (rkoron007) [inline website/data/docs-nav-data.json:2060]: Do these `href`s change?

**Review by rkoron007 — APPROVED** [review]
In general looks great! A couple of teeny questions and suggestions

**Review by krastin — APPROVED** [review]
Looking good! Just needed a little tweak for the nav data 🚀

**Pull request merged**